opal/resource.go

```diff
@@ -356,27 +469,40 @@ func resourceResourceRead(ctx context.Context, d *schema.ResourceData, m any) di
 		return diagFromErr(ctx, err)
 	}
 
+	requestConfigurations := make([]any, 0, len(resource.RequestConfigurationList))
+	for _, requestConfiguration := range resource.RequestConfigurationList {
+		requestConfigurationI, err := parseSDKRequestConfiguration(ctx, &requestConfiguration)
+		if err != nil {
+			return diagFromErr(ctx, err)
+		}
+		requestConfigurations = append(requestConfigurations, requestConfigurationI)
+	}
+
 	d.SetId(resource.ResourceId)
 	if err := multierror.Append(
 		d.Set("name", resource.Name),
 		d.Set("description", resource.Description),
 		d.Set("resource_type", resource.ResourceType),
 		d.Set("app_id", resource.AppId),
 		d.Set("admin_owner_id", resource.AdminOwnerId),
-		d.Set("auto_approval", resource.AutoApproval),
 		d.Set("require_mfa_to_approve", resource.RequireMfaToApprove),
 		d.Set("require_mfa_to_connect", resource.RequireMfaToConnect),
-		d.Set("require_mfa_to_request", resource.RequireMfaToRequest),
-		d.Set("require_support_ticket", resource.RequireSupportTicket),
-		d.Set("max_duration", resource.MaxDuration),
-		d.Set("recommended_duration", resource.RecommendedDuration),
-		d.Set("request_template_id", resource.RequestTemplateId),
-		d.Set("is_requestable", resource.IsRequestable),
+		d.Set("request_configuration", requestConfigurations),
 		// XXX: We don't get the metadata back. Will terraform state be okay?
 	); err.ErrorOrNil() != nil {
 		return diagFromErr(ctx, err)
 	}
 
+	if len(requestConfigurations) == 0 {
+		d.Set("auto_approval", resource.AutoApproval)
+		d.Set("require_mfa_to_request", resource.RequireMfaToRequest)
+		d.Set("require_support_ticket", resource.RequireSupportTicket)
+		d.Set("max_duration", resource.MaxDuration)
+		d.Set("recommended_duration", resource.RecommendedDuration)
+		d.Set("request_template_id", resource.RequestTemplateId)
+		d.Set("is_requestable", resource.IsRequestable)
+	}
```

Comment:
what's the plan for deprecating these (how long will we continue to support this? we should probably bump minor ver at least when this is deprecated)

Reply:
Gonna talk to Lance and Rish about the exact details on who is using these, but ideally we delete these ASAP since the code is quite annoying with both these as an option